### PR TITLE
Fix CC graceful downscale bug which happens if pod termination took t…

### PR DESCRIPTION
**Pull request by baluchicken**
…oo long

| Q               | A
| --------------- | ---
| Bug fix?        | yes|
| New feature?    | no|
| API breaks?     | no|
| Deprecations?   | no|
| License         | Apache 2.0

### What's in this PR?

This PR fixes the issue which happens when the pod termination takes too long, and a reconcile logic finds the pod which is already in terminating state. It causes Graceful Downscale error with Stuck CC.

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

File: config/base/crds/kafka.banzaicloud.io_kafkaclusters.yaml [hunk 1]
```diff
@@ -960,6 +960,16 @@ spec:
                   type: string
                 cruiseControlEndpoint:
                   type: string
+                cruiseControlTaskSpec:
+                  properties:
+                    RetrySleepSeconds:
+                      type: integer
+                    retryCount:
```
Comment (stoader) [inline config/base/crds/kafka.banzaicloud.io_kafkaclusters.yaml]: add description

File: config/base/crds/kafka.banzaicloud.io_kafkaclusters.yaml [hunk 2]
```diff
@@ -960,6 +960,16 @@ spec:
                   type: string
                 cruiseControlEndpoint:
                   type: string
+                cruiseControlTaskSpec:
+                  properties:
+                    RetrySleepSeconds:
```
Comment (stoader) [inline config/base/crds/kafka.banzaicloud.io_kafkaclusters.yaml]: add description

File: pkg/resources/kafka/kafka.go [hunk 3]
```diff
@@ -176,9 +177,58 @@ func (r *Reconciler) Reconcile(log logr.Logger) error {
 			deletedBrokers = append(deletedBrokers, pod)
 		}
 		for _, broker := range deletedBrokers {
-			err = scale.DownsizeCluster(broker.Labels["brokerId"], broker.Namespace, r.KafkaCluster.Spec.CruiseControlConfig.CruiseControlEndpoint, r.KafkaCluster.Name)
-			if err != nil {
-				log.Error(err, "graceful downscale failed.")
+			if broker.ObjectMeta.DeletionTimestamp != nil {
+				log.Info(fmt.Sprintf("Broker %s is already on terminating state", broker.Labels["brokerId"]))
```
Comment (stoader) [inline pkg/resources/kafka/kafka.go]: add cluster name to the log context

File: pkg/util/cruisecontrol/common.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cruisecontrol
+
+import "time"
+
+// ParseTimeStampToUnixTime parses the given CC timeStamp to time format
+func ParseTimeStampToUnixTime(timestamp string) (*time.Time, error) {
```
Comment (stoader) [inline pkg/util/cruisecontrol/common.go]: why do we need to return `*time.Time` instead of `time.Time`?

File: pkg/resources/kafka/kafka.go [hunk 5]
```diff
@@ -701,6 +642,57 @@ func (r *Reconciler) reconcileKafkaPod(log logr.Logger, desiredPod *corev1.Pod)
 
 }
 
+func (r *Reconciler) checkCCTaskState(brokerId string, brokerState v1beta1.BrokerState, log logr.Logger) error {
```
Comment (stoader) [inline pkg/resources/kafka/kafka.go]: add `brokerId` to log context

File: pkg/resources/kafka/kafka.go (hunk 3 above)
Comment (baluchicken) [reply pkg/resources/kafka/kafka.go]: Added log context

**Pull request merged**